src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -1202,19 +1202,81 @@ private async Task<Size> CalculateStorageSize(string databaseName)
         public async Task SetUnusedDatabaseIds()
         {
             var database = GetStringQueryString("name");
+            var validate = GetBoolValueQueryString("validate", required: false);
+
             await ServerStore.EnsureNotPassiveAsync();
 
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
             using (var json = await context.ReadForDiskAsync(RequestBodyStream(), "unused-databases-ids"))
             {
                 var parameters = JsonDeserializationServer.Parameters.UnusedDatabaseParameters(json);
+                if (validate.HasValue && validate.Value)
+                {
+                    var token = CreateHttpRequestBoundTimeLimitedOperationToken(ServerStore.Configuration.Cluster.OperationTimeout.AsTimeSpan);
+                    await ValidateUnusedIdsAsync(parameters.DatabaseIds, database, token.Token);
+                }
+
                 var command = new UpdateUnusedDatabaseIdsCommand(database, parameters.DatabaseIds, GetRaftRequestIdFromQuery());
                 await ServerStore.SendToLeaderAsync(command);
             }
 
             NoContentStatus();
         }
 
+        private async Task ValidateUnusedIdsAsync(IEnumerable<string> unusedIds, string databaseName, CancellationToken token)
```

Comment:
I think it is much cleaner and more efficient to make `unusedIds` be a `Hashset`

Reply:
Done